browser-switch/src/main/java/com/braintreepayments/api/BrowserSwitchClient.java

```diff
@@ -167,6 +167,40 @@ public BrowserSwitchResult getResult(@NonNull FragmentActivity activity) {
         return result;
     }
 
+    /**
+     * Parses and returns a browser switch result if a match is found.
+     *
+     * @param context     The context used to check for pending browser switch requests
+     * @param requestCode The request code for the matching pending request
+     * @param intent      Intent to evaluate for deep link result
+     * @return
+     */
+    @Nullable
+    public BrowserSwitchResult parseResult(@NonNull Context context, int requestCode, @Nullable Intent intent) {
```

Comment:
When should a merchant use `parseResult` over `deliverResult`?

Reply:
Gotcha, that would be helpful 

Maybe we can clarify in the `parseResult` docstring that it doesn't modify persistent storage in the same way as `deliver`

Reply:
Yeah I like that idea. I can update the docstring to call it out.